src/pybind/mgr/prometheus/module.py

```diff
@@ -512,10 +517,47 @@ def collect(self):
 
         return self.metrics.metrics
 
+    def get_file_sd_config(self):
+        servers = self.list_servers()
+        targets = []
+        for server in servers:
+            hostname = server.get('hostname', '')
+            for service in server.get('services', []):
+                if service['type'] != 'mgr':
+                    continue
+                id_ = service['id']
+                # get port for prometheus module at mgr with id_
+                # TODO use get_config_prefix or get_config here once
+                # https://github.com/ceph/ceph/pull/20458 is merged
+                result = CommandResult("")
+                global_instance().send_command(
+                    result, "mon", '',
+                    json.dumps({
+                        "prefix": "config-key get",
+                        'key': "mgr/prometheus/{}/server_port".format(id_),
+                    }),
+                                               "")
+                r, outb, outs = result.wait()
+                if r != 0:
+                    global_instance().log.error("Failed to retrieve port for mgr {}: {}".format(mgr, outs))
+                else:
+                    port = json.loads(outb)
+                    targets.append('{}:{}'.format(hostname, port))
+
+        ret = [
+            {
+                "targets": targets,
+                "labels": {}
```

Comment:
Maybe I am missing something but why are there these empty labels?

Reply:
This is just the prometheus file_sd_config format. I thought prometheus would not like not having the label key, but it turns out that it start fine when this key is missing (at least with version 2.1.0). As this doesn't add anyhting I'm happy to remove it.

I guess an argument could be made for a user clearly seeing where a per-target label config would be added, but again: happy to do without it.

Reply:
Ah, ok, that makes sense, with the explanation I am fine either way (having/not having the labels key in the dict).